### PR TITLE
Linker script modified for R5030

### PR DESCRIPTION
**Pull request by bqam-ublox**
Linker script modified for R5030: r5030.ld

INITIAL_SP defined in mbed_rtx.h

Notes:
* Pull requests will not be accepted until the submitter has agreed to the [contributer agreement](https://github.com/ARMmbed/mbed-os/blob/master/CONTRIBUTING.md).
* This is just a template, so feel free to use/remove the unnecessary things

## Description
A few sentences describing the overall goals of the pull request's commits.

## Status
**READY/IN DEVELOPMENT/HOLD**

## Migrations
If this PR changes any APIs or behaviors, give a short description of what *API users* should do when this PR is merged.

YES | NO

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()

## Todos
- [ ] Tests
- [ ] Documentation

## Deploy notes
Notes regarding the deployment of this PR. These should note any
required changes in the build environment, tools, compilers, etc.

## Steps to test or reproduce
Outline the steps to test or reproduce the PR here.

### CODE REVIEW

File: targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct [hunk 1]
```diff
@@ -22,16 +22,16 @@
  *************************************************************
  */
 
-LR_IROM1 0x00000000 0x00040000  {    ; load region size_region
+; R5030_APPS_PROC: 128 KB FLASH (0x200000) + 1024 KB SRAM (0x100000)
```
Comment (RobMeades) [inline targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct:25]: Good to have this comment: I guess the FLASH portion is not mapped in yet (as I can't see anything of size `0x200000` below)?  Or did you mean `0x20000`?

File: targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct [hunk 2]
```diff
@@ -22,16 +22,16 @@
...
+LR_IROM1 0x00000000 0x20000  {    ; load region size_region
   ER_IROM1 0x00000000 0x00040000  {  ; load address = execution address
```
Comment (RobMeades) [inline targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct:27]: Which memory area is this?

File: targets/TARGET_ublox/TARGET_R5030/device/cmsis_nvic.h [hunk 3]
```diff
@@ -18,7 +18,12 @@
 #ifndef MBED_CMSIS_NVIC_H
 #define MBED_CMSIS_NVIC_H
 
-#define NVIC_NUM_VECTORS      (16 + 32)   // CORE + MCU Peripherals
+
+// CORE: 16 vectors = 64 bytes from 0x00 to 0x3F
+// MCU Peripherals: 45 vectors =  180 bytes from 0x40 to 0x6D
```
Comment (RobMeades) [inline targets/TARGET_ublox/TARGET_R5030/device/cmsis_nvic.h]: I make 180 bytes to be from 0x40 to 0xF4.

File: targets/TARGET_ublox/TARGET_R5030/device/cmsis_nvic.h [hunk 4]
```diff
@@ -18,7 +18,12 @@
...
+// Total: 61 vectors = 244 bytes to be reserved in RAM
```
Comment (RobMeades) [inline targets/TARGET_ublox/TARGET_R5030/device/cmsis_nvic.h]: You have 244 (`0xF4`) bytes according to this comment and `NVIC_NUM_VECTORS` but in the linker files above you use `0xF8`.  Which is correct?

File: targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct (hunk 1 above)
Comment (bqam-ublox) [reply targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct:25]: Yes. Its a mistake in the comment. Its 0x20000

File: targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct (hunk 2 above)
Comment (bqam-ublox) [reply targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct:27]: ER_ROM1 is the first execution region. However, its size is same as of load region in this case. Corrected to 0x20000.

File: targets/TARGET_ublox/mbed_rtx.h [hunk 5]
```diff
@@ -25,7 +25,8 @@
 
 #if defined(TARGET_R5030)
 #ifndef INITIAL_SP
-#define INITIAL_SP              (0x01000000 + 0x05000 - 256)
+#define INITIAL_SP              (0x36000000)
```
Comment (RobMeades) [inline targets/TARGET_ublox/mbed_rtx.h]: If this value is correct you should add a comment explaining how it is calculated.  If RAM starts at `0x20000000` and is `0x100000` big, then `0x36000000` is beyond the end of the RAM.

File: targets/TARGET_ublox/TARGET_R5030/device/cmsis_nvic.h (hunk 3 above)
Comment (bqam-ublox) [reply targets/TARGET_ublox/TARGET_R5030/device/cmsis_nvic.h]: Corrected. Also, there are 62 vectors in total now.

File: targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct (hunk 1 above)
Comment (bqam-ublox) [reply targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct:25]: @RobMeades changes have been made in another commit. Kindly review when possible. Thanks
(https://github.com/fahim-ublox/mbed-os/pull/3/commits/202a7c5edab21030cc5f68baa7bbd35ee1c3011f)

File: targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct (hunk 2 above)
Comment (RobMeades) [reply targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct:27]: This bit looks good.

File: targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct (hunk 1 above)
Comment (RobMeades) [reply targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct:25]: This bit looks good.

Comment (bqam-ublox) [reply targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_ARM_STD/r5030.sct:25]: Alright, kindly approve if there is no other discrepancy. Thanks

File: targets/TARGET_ublox/mbed_rtx.h (hunk 5 above)
Comment (RobMeades) [reply targets/TARGET_ublox/mbed_rtx.h]: This review comment still remains I think.  Is `0x36000000` really correct, if so why?

File: targets/TARGET_ublox/mbed_rtx.h [hunk 6]
```diff
@@ -25,7 +25,8 @@
 
 #if defined(TARGET_R5030)
 #ifndef INITIAL_SP
-#define INITIAL_SP              (0x01000000 + 0x05000 - 256)
+#define INITIAL_SP              (0x20000000 + 0x100000)
```
Comment (RobMeades) [inline targets/TARGET_ublox/mbed_rtx.h:28]: Looks good to me :-).

**Review by RobMeades — APPROVED** [review]
LGTM.

**Pull request merged**